linalg/ode.hpp

```diff
@@ -92,26 +170,36 @@ class ODESolver
       while (t < tf) { Step(x, t, dt); }
    }
 
-   /// Function for getting and setting the state vectors
-   virtual int   GetMaxStateSize() { return 0; }
-   virtual int   GetStateSize() { return 0; }
-   virtual const Vector &GetStateVector(int i)
-   {
-      mfem_error("ODESolver has no state vectors");
-      Vector *s = NULL; return *s; // Make some compiler happy
-   }
-   virtual void  GetStateVector(int i, Vector &state)
-   {
-      mfem_error("ODESolver has no state vectors");
-   }
-   virtual void  SetStateVector(int i, Vector &state)
-   {
-      mfem_error("ODESolver has no state vectors");
-   }
```

Comment:
Can we just deprecate these for now, instead of deleting them? This way we are backward compatible for some time.

Also, it is good to mention the changing interface in CHANGELOG.

Reply:
Now I see that these no longer implementable in this class. In this case, we should definitely have this mentioned in the CHANGELOG.

Reply:
Also, mention the replacement interface available in the sub-class `ODESolverWithStates`.

Reply:
Will add to CHANGELOG.

Reply:
Update the changelog